coins/bitcoin/src/rpc.rs

```diff
@@ -0,0 +1,538 @@
+use anyhow::Result;
+use serde::{de::DeserializeOwned};
+use std::{collections::HashMap, fmt::Debug, vec::Vec, str::FromStr};
+
+use bitcoin::{
+    hashes::hex::FromHex, secp256k1::ecdsa::Signature, Address, Amount, EcdsaSighashType, OutPoint,
+    PrivateKey, Transaction, BlockHash,
+};
+
+use crate::rpc_helper::{
+    empty_arr, empty_obj, handle_defaults, into_json, null, opt_into_json, JsonOutPoint, RawTx,
+    RpcConnectionError, RpcParams, RpcResponse, GetBlockWithDetailResult
+};
+
+use bitcoincore_rpc_json::{
+    AddressType, CreateRawTransactionInput, EstimateSmartFeeResult, FundRawTransactionOptions,
+    FundRawTransactionResult, GetBlockResult, GetRawTransactionResult, GetTransactionResult,
+    ListUnspentResultEntry, LoadWalletResult, SignRawTransactionInput, SignRawTransactionResult,
+    ListTransactionResult, TestMempoolAcceptResult
+};
+
+#[derive(Debug, Clone)]
+pub struct Rpc {
+    url: String,
+}
+
+impl Rpc {
+    pub fn new(url: String, username: String, userpass: String) -> anyhow::Result<Rpc> {
+        Ok(Rpc {
+            url: format!("http://{}:{}@{}", username, userpass, url),
+        })
+    }
+
+    pub async fn rpc_call<Response: DeserializeOwned + Debug>(
+        &self,
+        method: &str,
+        params: &[serde_json::Value],
+    ) -> anyhow::Result<Response> {
+        let client = reqwest::Client::new();
+        //println!("METHOD : ---------->");
+        //dbg!(&method);
+        //dbg!(&params);
+        let res = client
+            .post(&self.url)
+            .json(&RpcParams {
+                jsonrpc: "2.0",
+                id: (),
+                method,
+                params,
+            })
+            .send()
+            .await?
+            .text()
+            .await?;
+        //dbg!(&res);
+        let parsed_res: RpcResponse<Response> = serde_json::from_str(&res)
+            .map_err(|_| anyhow::Error::new(RpcConnectionError::ParsingError))?;
+        //dbg!(&parsed_res.error);
+        match parsed_res.error {
+            None => Ok(parsed_res.result.unwrap()),
+            Some(r) => Err(anyhow::Error::msg(r.message)), //Err(anyhow::Error::new(RpcConnectionError::ResultError(r.message))),
+        }
+    }
+
+    pub async fn get_height(&self) -> anyhow::Result<usize> {
+        let info: usize = self.rpc_call::<usize>("getblockcount", &[]).await?;
+        Ok(info)
+    }
+
+    pub async fn get_block_info(&self, block_hash: &str) -> anyhow::Result<GetBlockResult> {
```

Comment:
Remove, since we only use it to get the block number and I'd like to minimize this file please.